docs/source/en/basics/middleware.md

```diff
@@ -0,0 +1,202 @@
+title: Middleware
+---
+
+In [the previous chapter](../intro/egg-and-koa.md), we say that Egg is based on Koa 1, so the form of middleware in Egg is the same as in Koa 1, i.e. they are both based on [the onion model](../intro/egg-and-koa.md#midlleware) of generator function.
+
+## Writing Middleware
+
+### how to write
+
+We begin by writing a simple gzip middleware, to see how to write middleware
+
+```js
+const isJSON = require('koa-is-json');
+const zlib = require('zlib');
+
+function* gzip(next) {
+  yield next;
+
+  // convert the reaponse body to gzip after the completion of the execution of subsequent middleware
+  let body = this.body;
+  if (!body) return;
+  if (isJSON(body)) body = JSON.stringify(body);
+
+  // set gzip body, correct the reponse header
+  const stream = zlib.createGzip();
+  stream.end(body);
+  this.body = stream;
+  this.set('Content-Encoding', 'gzip');
+}
+```
+
+You might find that the middleware's writing style in the framework is exactly the same as in Koa, so any middleware in Koa can be used directly by the framework.
+
+### Configuration
+
+Usually the middleware has its own configuration. In the framework, a complete middleware is including the configuration process. We agree that a middleware is a separate file placed in `app/middleware` directory, which needs a exports function that take two paramters:
+
+- options: the configuration field of the middleware, `app.config[${middlewareName}]` will be passed in by the frame
```

Comment:
👍 You're right, it's my fault. 

Next time I will paste the doc to LibreOffice (any recommendation?) to find typos before commit it.